source/engine/graphics/shaders/EnvironmentDataShaderResource.h

```diff
@@ -1,16 +1,16 @@
 #pragma once
 
-#include "ShaderResourceService.h"
-#include "ShaderDescriptorSets.h"
-#include "UniformBuffer.h"
+#include "graphics/buffers/UniformBuffer.h"
+#include "graphics/shaders/ShaderResourceService.h"
+#include "graphics/shaders/ShaderDescriptorSets.h"
 
 namespace nc::graphics
 {
-    class EnvironmentDataManager : public IShaderResourceService<EnvironmentData>
+    class EnvironmentDataShaderResource : public IShaderResourceService<EnvironmentData>
```

Comment:
To clarify my mental model of these guys: is a shader resource conceptually the whole buffer, rather than the individual items in the buffer? If so, I think this rename makes things much clearer.

What do you think about renaming `IShaderResourceService` to `IShaderResource`? There is currently a using alias in the mix named `ShaderResourceService`, whose purpose is to handle the service locator side of things. The interface doesn't directly have anything to do with services.

Throwing the rest of my comments here as I can't comment on files that were just moved:
- The types in `shaders/Environment.h` are not in the graphics namespace.
- `graphics/SceneNavigationCamera.cpp` should also go in `components`.

Reply:
Building off of the first and second items: There is also a concept named `ShaderResource` which maps to the individual types, not buffers. Assuming the answer to my first question is yes, then `ShaderResource` is an awkward name for the concept. I don't know what to call it. Maybe you have ideas, or we can discuss.